### PR TITLE
library: Ported `ToggleButton` to RUST

### PR DESCRIPTION
**Pull request by onkarrai06**
Ported ToggleButton to RUST via reference of JS implementation. Converted the dictionary in JS to a Hashmap and cloned it in order to avoid borrowing issues. Please check and let me know of any anomaly.

### CODE REVIEW

File: src/Library/demos/Toggle Button/code.rs [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+    let buttons: HashMap<&str, &str> = [
+        ("button_no_look", "Don't look"),
+        ("button_look", "Look"),
+        ("button_camera", "Camera"),
+        ("button_flashlight", "Flashlight"),
+        ("button_console", "Console"),
+    ]
+    .iter()
+    .cloned()
+    .collect();
```
Comment (Hofer-Julian) [inline src/Library/demos/Toggle Button/code.rs:15]: You can use `from` instead: https://doc.rust-lang.org/std/collections/hash_map/struct.HashMap.html#method.from

File: src/Library/demos/Toggle Button/code.rs [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
+    for (id, name) in buttons.iter() {
+        let button: gtk::ToggleButton = workbench::builder().object(*id).unwrap();
```
Comment (Hofer-Julian) [inline src/Library/demos/Toggle Button/code.rs:18]:
```suggestion
    for (id, name) in buttons.into_iter() {
        let button: gtk::ToggleButton = workbench::builder().object(id).unwrap();
```

File: src/Library/demos/Toggle Button/code.rs [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+        let cloned_name = name.to_string();
+        button.connect_active_notify(move |button| {
+            let status = if button.is_active() { "On" } else { "Off" };
+            println!("{} {}", cloned_name, status);
```
Comment (Hofer-Julian) [inline src/Library/demos/Toggle Button/code.rs:22]:
```suggestion
            println!("{cloned_name} {status}");
```

**Comment by onkarrai06** [comment]
Updated, please check and let me know of any problem.

File: src/Library/demos/Toggle Button/code.rs [hunk 4]
```diff
@@ -0,0 +1,22 @@
+use crate::workbench;
+use gtk::prelude::*;
+use std::collections::HashMap;
+
+pub fn main() {
+    let buttons: HashMap<&str, &str> = HashMap::from([
```
Comment (Hofer-Julian) [inline src/Library/demos/Toggle Button/code.rs:6]: The type annotation should not be needed

Comment (onkarrai06) [reply src/Library/demos/Toggle Button/code.rs:6]: For some reason, it did not work for me without it. Can you help me why? Maybe its just a problem on my side.

Comment (Hofer-Julian) [reply src/Library/demos/Toggle Button/code.rs:6]: What is the error message you get?

File: src/Library/demos/Toggle Button/code.rs [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+        ("button_no_look", "Don't look"),
+        ("button_look", "Look"),
+        ("button_camera", "Camera"),
+        ("button_flashlight", "Flashlight"),
+        ("button_console", "Console"),
+    ]);
+
+    for (id, name) in buttons.into_iter() {
+        let button: gtk::ToggleButton = workbench::builder().object(id).unwrap();
+        let cloned_name = name.to_string();
```
Comment (saadulkh) [inline src/Library/demos/Toggle Button/code.rs:16]: This variable can be omitted. Use `name` directly.

**Comment by Hofer-Julian** [comment]
I've removed the annotation and followed @saadulkh's helpful suggestion in https://github.com/sonnyp/Workbench/pull/562/commits/5cde951d10fb9fe4d136f16dca2b02415fd77fb0

**Review by Hofer-Julian — APPROVED** [review]

**Comment by onkarrai06** [comment]
Thank you for both of your suggestions and help!

**Pull request merged**